rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -460,6 +460,14 @@ void SqliteStorage::create_topic(
 {
   std::lock_guard<std::mutex> db_lock(database_write_mutex_);
   if (topics_.find(topic.name) == std::end(topics_)) {
+    std::vector<rosbag2_storage::Rosbag2QoS> to_encode;
+    to_encode.reserve(topic.offered_qos_profiles.size());
+    std::transform(
+      topic.offered_qos_profiles.begin(),
+      topic.offered_qos_profiles.end(),
+      std::back_inserter(to_encode),
+      [](auto & qos) {return static_cast<rosbag2_storage::Rosbag2QoS>(qos);});
+    auto yaml_node = YAML::convert<std::vector<rosbag2_storage::Rosbag2QoS>>::encode(to_encode);
```

Comment:
Here, a conversion from `std::vector<rclcpp::QoS>` to serialized YAML (`std::string`) is needed.